src/topology/fanout.rs

```diff
@@ -0,0 +1,277 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use futures::FutureExt;
+
+use crate::bounded_channel::BoundedSender;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// A fanout component that distributes messages to multiple consumers.
+///
+/// The fanout component takes a message and sends it to all configured consumers.
+/// The message is cloned for all consumers except the last one to avoid unnecessary cloning.
+///
+pub struct Fanout<T> {
+    consumers: Vec<BoundedSender<Vec<T>>>,
```

Comment:
Should these BoundedSenders required a Vec<T> or just a T and allow the user to decide?